addons/hr_timesheet/models/project_task.py

```diff
@@ -174,6 +174,8 @@ def action_view_subtask_timesheet(self):
         graph_view_id = self.env.ref("hr_timesheet.view_hr_timesheet_line_graph_by_employee").id
         new_views = []
         for view in action['views']:
+            if (not is_internal_user or self.env.context.get('is_project_sharing', False) and view[1] not in ['tree', 'kanban', 'form']):
```

Comment:
You did not correctly apply my suggestion... 😕 

```suggestion
            if (not is_internal_user or self.env.context.get('is_project_sharing', False)) and view[1] not in ['tree', 'kanban', 'form']:
```

Reply:
Also, no need to give a default value inside `get` method of the context dict.

```suggestion
            if (not is_internal_user or self.env.context.get('is_project_sharing')) and view[1] not in ['tree', 'kanban', 'form']:
```